lib/vsc/mympirun/mpi/openmpi.py

```diff
@@ -66,7 +69,23 @@ def pinning_override(self):
         self.log.debug("pinning_override: type %s ", override_type)
 
         ranktxt = ""
-        sockets_per_node = 2
+
+        sockets_per_node = self.options.sockets
+        if sockets_per_node == 0:
+            try:
+                proc_cpuinfo = open('/proc/cpuinfo').read()
+            except IOError:
+                self.log.error("Number of sockets per node not found: please use --sockets-per-node")
+                sys.exit(1)
+
+            if proc_cpuinfo:
+                res = re.findall('^physical id.*', proc_cpuinfo, re.M)
+                sockets_per_node = len(nub(res))
+                if sockets_per_node == 0:
+                    self.log.error("Something went wrong: found sockets per node = 0. Please use --sockets-per-node")
+                    sys.exit(1)
+                self.log.debug("Sockets per node found in cpuinfo: set to %s" % sockets_per_node)
```

Comment:
@Caylo let's put this in a separate method, this `pinning_override` method is getting too big with this included

```python
def determine_sockets_per_node(self):
   """Useful docstring here"""
   ...
   return sockets_per_node

def pinning_override(self):
    ...
    sockets_per_node = self.determine_sockets_per_node()
```

Can we come up with a way to test this btw?